src/api/api.c

```diff
@@ -1491,16 +1489,259 @@ void getGateway(const int *sock)
+		snprintf(fname, sizeof(fname)-1, "/sys/class/net/%s/speed", new->name);
+		if((f = fopen(fname, "r")) == NULL || fscanf(f, "%i", &(new->speed)) != 1)
+			new->speed = -1;
+
+		// Get total transmitted bytes
+		snprintf(fname, sizeof(fname)-1, "/sys/class/net/%s/statistics/tx_bytes", new->name);
+		if((f = fopen(fname, "r")) == NULL || fscanf(f, "%zi", &(new->tx_bytes)) != 1)
+			new->tx_bytes = -1;
+
+		// Get total transmitted bytes
+		snprintf(fname, sizeof(fname)-1, "/sys/class/net/%s/statistics/rx_bytes", new->name);
+		if((f = fopen(fname, "r")) == NULL || fscanf(f, "%zi", &(new->rx_bytes)) != 1)
+			new->rx_bytes = -1;
+
+		// Get IP address(es) of this interface
+		if(ifap)
+		{
+			// Walk through linked list of interface addresses
+
+			for(struct ifaddrs *ifa = ifap; ifa != NULL; ifa = ifa->ifa_next)
+			{
+				// Skip interfaces without an address and those
+				// not matching the current interface
+				if(ifa->ifa_addr == NULL || strcmp(ifa->ifa_name, new->name) != 0)
+					continue;
+
+				// If we reach this point, we found the correct interface
+				new->family = ifa->ifa_addr->sa_family;
+				char host[NI_MAXHOST] = { 0 };
+				if(new->family == AF_INET || new->family == AF_INET6)
+				{
+					// Get IP address
+					const int s = getnameinfo(ifa->ifa_addr,
+					                          (new->family == AF_INET) ?
+					                               sizeof(struct sockaddr_in) :
+					                               sizeof(struct sockaddr_in6),
+					                          host, NI_MAXHOST,
+					                          NULL, 0, NI_NUMERICHOST);
+					if (s != 0)
+					{
+						logg("API warning: getnameinfo() failed: %s\n", gai_strerror(s));
+						continue;
+					}
+
+					if(new->family == AF_INET)
+					{
+						// IPv4 address
+						if(!new->ip.v4)
+						{
+							// First or only IPv4 address of this interface
+							new->ip.v4 = strdup(host);
+						}
+						else
+						{
+							// Create comma-separated list
+							char *new_v4 = calloc(strlen(new->ip.v4) + strlen(host) + 2, sizeof(char));
+							sprintf(new_v4, "%s,%s", new->ip.v4, host);
+							free(new->ip.v4);
+							new->ip.v4 = new_v4;
+						}
+					}
+					else if(new->family == AF_INET6)
+					{
+						// IPv6 address
+						if(!new->ip.v6)
+						{
+							// First or only IPv6 address of this interface
+							new->ip.v6 = strdup(host);
+						}
+						else
+						{
+							// Create comma-separated list
+							char *new_v6 = calloc(strlen(new->ip.v6) + strlen(host) + 2, sizeof(char));
+							sprintf(new_v6, "%s,%s", new->ip.v6, host);
+							free(new->ip.v6);
+							new->ip.v6 = new_v6;
+						}
+					}
+				}
+			}
+		}
+
+		// Add to end of the linked list
+		if(!*head)
+			*head = new;
+		if(tail)
+			tail->next = new;
+		tail = new;
+
+		tx_sum += new->tx_bytes;
+		rx_sum += new->rx_bytes;
+	}
+
+	freeifaddrs(ifap);
+
+	// Create sum entry only if there is more than one interface
```

Comment:
I would not create sum only if more than one interface is found. The API output should stay constant, even if the number of interfaces changes. Users might search for `sum` and wonder why their scripts fail if only one interface is up.

Reply:
It's not about interfaces being up but existing. You should always have at least `lo` in addition to your normal interface. The current simplification is to keep the code a bit simpler.